lib/pure/collections/tableimpl.nim

```diff
@@ -124,9 +124,9 @@ template initImpl(result: typed, size: int) =
   when ctAnd(declared(SharedTable), type(result) is SharedTable):
     init(result, size)
   else:
-    assert isPowerOfTwo(size)
+    var correctSize = rightSize(size)
```

Comment:
can you also change other code that uses ` assert isPowerOfTwo(initialSize)`, eg in sharedtables + other files?

Reply:
> eg in sharedtables + other files

What other files?

Reply:
> used only in 1 branch but not the other

When is that other branch used, btw?

I've now quickly skimmed through `sharedtables.nim` and I don't see any calls of `initImpl` :/

Reply:
> What other files?

```
lib/pure/collections/deques.nim:70:10:  assert isPowerOfTwo(initialSize)
lib/pure/collections/setimpl.nim:19:10:  assert isPowerOfTwo(size)
lib/pure/collections/sharedtables.nim:218:10:  assert isPowerOfTwo(initialSize)
lib/pure/collections/tableimpl.nim:127:12:    assert isPowerOfTwo(size)
```

Reply:
> I've now quickly skimmed through sharedtables.nim and I don't see any calls of initImpl :/

it may be indirect; it was definitely being used when i edited that part of the code; in any case if it's not being used, a `static: doAssert false` would make sure we're not going there by accident in some future update

Reply:
`nim r tests/stdlib/tsharedtable.nim` would indeed crash when using `static: doAssert false` here